### PR TITLE
[next][SuperH] Fix missing setting detail->sh

### LINKED ISSUE #2424: SuperH missing setting detail->sh on disassemble
### Work environment

| Questions                                | Answers
|------------------------------------------|--------------------
| OS/arch/bits                             |  Ubuntu x86 64
| Architecture                             |  sh
| Source of Capstone                       | `git clone`
| Version/git commit                       | (v5.0.1, 1bd2a32f2afc6e08225d7491b82619cf4c8e87d9), (next, 4f964a264ec25eb9d468a5495fbd6142778c3a47)

### Instruction bytes giving faulty results

All instructions with operands have wrong detailed output as `detail->sh` never gets set.
```
0x32,0x11,0x92,0x00
```

### Expected results

It should include operands information.

### Steps to get the wrong result

With `cstool`:

```sh
cstool/cstool -d sh2a '32119200'
 0  32 11 92 00  movu.w @(1024,r1),r2
        ID: 100 (movu)
        reading from reg: r1
        writing to reg:   r2
        groups_count: 0
```

Expected output should be:
```
cstool/cstool -d sh2a '32119200'
 0  32 11 92 00  movu.w @(1024,r1),r2
        ID: 100 (movu)
        reading from reg: r1
        writing to reg:   r2
        groups_count: 0
        op_count: 2
                operands[0].type: MEM
                        operands[0].mem.reg: REG = r1
                        operands[0].mem.disp: 0x400
                address mode: Register Indirect with Predecrement
                operands[1].type: REG = r2
```

This bug exists on both `v5` and `next` branches. Will this issue be resolved after the modern testing #2384? This bug makes me feel the SuperH arch is almost not tested at all.

### PR DESCRIPTION
**Pull request by david942j**
detail->sh is never set. Which makes the detailed output related to operands is always missing.

 

**Your checklist for this pull request**
- [x] I've documented or updated the documentation of every API function and struct this PR changes.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)

**Detailed description**

`detail->sh` is never set. Which makes the detailed output related to operands is always missing.

**Test plan**

`build/suite/cstest/cstest tests/issues/issues.yaml`, the newly added case passes/fails with/without the change.

**Closing issues**

closes #2424 (along with #2465)

### CODE REVIEW

File: tests/test_arm64
Comment (Rot127) [inline tests/test_arm64:1]: Please remove the binary

**Review by Rot127 — CHANGES_REQUESTED** [review]
Thanks a lot! Really weird it hasn't been discovered before.

Please delete the binary.
The failing clang-tidy test is because there are unrelated problems.
I asked @lhsazevedo in https://github.com/capstone-engine/capstone/pull/2389 already to fix it.
I you want to you can of course open a PR to his branch to speed things up.

File: tests/test_arm64
Comment (david942j) [reply tests/test_arm64:1]: Oops. Done.

**Review by Rot127 — APPROVED** [review]

**Comment by Rot127** [comment]
Please rebase though and fix the conflicts. The clang-tidy job can be ignored. It is fixed in the coverity PR.

**Comment by Rot127** [comment]
@david942j The clang-tidy issues are fixed in https://github.com/capstone-engine/capstone/pull/2469. But I cannot merge, so we have to wait until @kabeor finds time to review and merges stuff.

**Comment by david942j** [comment]
@kabeor I just resolved conflicts with the latest next branch, this PR is ready to be reviewed, thanks!

**Review by Rot127 — APPROVED** [review]

**Review by kabeor — APPROVED** [review]
Cool, thanks.

**Pull request merged**
